src/Classes/GemSelectControl.lua

```diff
@@ -512,6 +512,18 @@ function GemSelectClass:AddCommonGemInfo(gemInstance, grantedEffect, addReq, mer
 		if grantedEffectLevel.cost.Life then
 			self.tooltip:AddLine(16, string.format("^x7F7F7FLife Cost: ^7%d", grantedEffectLevel.cost.Life))
 		end
+		if grantedEffectLevel.cost.ES then
+			self.tooltip:AddLine(16, string.format("^x7F7F7FEnergy Shield Cost: ^7%d", grantedEffectLevel.cost.ES))
+		end
+		if grantedEffectLevel.cost.Rage then
+			self.tooltip:AddLine(16, string.format("^x7F7F7FRage Cost: ^7%d", grantedEffectLevel.cost.Rage))
+		end
+		if grantedEffectLevel.cost.ManaPercent then
+			self.tooltip:AddLine(16, string.format("^x7F7F7FMana Cost: ^7%d%%", grantedEffectLevel.cost.ManaPercent))
+		end
+		if grantedEffectLevel.cost.LifePercent then
+			self.tooltip:AddLine(16, string.format("^x7F7F7FLife Cost: ^7%d%%", grantedEffectLevel.cost.LifePercent))
+		end
```

Comment:
Think it's worth making this generic and use the `CostType` table to get the text to print?

Reply:
A global table may have some use but the string formatting is different in every place. There are still some minor things that need polishing so I'll create an issue to collect them